web/rest/brand/src/Controller/Provider/PutBillableCallRatingAction.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+namespace Controller\Provider;
+
+use ApiPlatform\Core\Exception\ResourceClassNotFoundException;
+use Ivoz\Provider\Domain\Model\Administrator\AdministratorInterface;
+use Ivoz\Provider\Domain\Model\BillableCall\BillableCall;
+use Ivoz\Provider\Domain\Model\BillableCall\BillableCallDto;
+use Ivoz\Provider\Domain\Model\Brand\BrandInterface;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorage;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Serializer\Serializer;
+use Ivoz\Provider\Domain\Model\BillableCall\BillableCallRepository;
+
+class PutBillableCallRatingAction
+{
+    protected $tokenStorage;
+    protected $serializer;
+    protected $requestStack;
+    protected $billableCallRepository;
+
+    public function __construct(
+        TokenStorage $tokenStorage,
+        Serializer $serializer,
+        RequestStack $requestStack,
+        BillableCallRepository $billableCallRepository
+    ) {
+        $this->tokenStorage = $tokenStorage;
+        $this->serializer = $serializer;
+        $this->requestStack = $requestStack;
+        $this->billableCallRepository = $billableCallRepository;
+    }
+
+    public function __invoke()
+    {
+        /** @var TokenInterface $token */
+        $token = $this->tokenStorage->getToken();
+
+        if (!$token || !$token->getUser()) {
+            throw new ResourceClassNotFoundException('User not found');
+        }
+
+        /** @var AdministratorInterface $user */
+        $user = $token->getUser();
+        /** @var BrandInterface $brand */
+        $brand = $user->getBrand();
+        if (!$brand) {
+            throw new ResourceClassNotFoundException('User brand not found');
+        }
+
+        $request = $this->requestStack->getCurrentRequest();
+        $content = $request->getContent();
+        $format = $request->getRequestFormat();
+
+        $data = $this->serializer->decode(
+            $content,
+            $format,
+            []
+        );
+        $data['ratingPlanName'] = 'Externally rated';
+
+        $callid = $request->attributes->get('callid');
+
+        $calls = $this->billableCallRepository->findByCallid(
+            $callid,
+            $brand->getId()
+        );
+
+        if (empty($calls)) {
+            throw new NotFoundHttpException();
+        }
+
+        if (count($calls) !== 1) {
+            $errorMsg = sprintf(
+                'Multiple calls found with callid %s',
+                $callid
+            );
+
+            throw new \DomainException($errorMsg, 500);
+        }
```

Comment:
If we leave it as is, it would be impossible to tarificate bounced calls within the same brand.
Maybe we should add extra conditions to avoid this scenario (maybe _direction_ with _outgoing_ as default value?)

Reply:
Updated repository query, thanks :)